From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ByThePowerOfScience
 <16433721+ByThePowerOfScience@users.noreply.github.com>
Date: Fri, 23 Feb 2024 15:45:33 -0500
Subject: [PATCH] 6249: Fixed false error in GhidraJarBuilder.java

Changed from "only giving an error if we WERE able to write to the sources jar" to "only giving an error if we were NOT able to write to the sources jar".

Moved zip.close out of if-else tree.
---
 .../Base/src/main/java/ghidra/util/GhidraJarBuilder.java | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/Ghidra/Features/Base/src/main/java/ghidra/util/GhidraJarBuilder.java b/Ghidra/Features/Base/src/main/java/ghidra/util/GhidraJarBuilder.java
index ef2ade34fe..71e9e72490 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/util/GhidraJarBuilder.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/util/GhidraJarBuilder.java
@@ -338,12 +338,11 @@ public class GhidraJarBuilder implements GhidraLaunchable {
 				wroteToZip |= writeZipRecursively(zip, srcDir.getAbsolutePath(), srcDir);
 			}
 		}
-		if (wroteToZip) {
-			System.out
-					.println("Can't create source zip!  Has source been downloaded and installed?");
-			// zip.close reports error if nothing has been written to it
-			zip.close();
+		if (!wroteToZip) {
+			System.out.println("Can't create source zip!  Has source been downloaded and installed?");
 		}
+		// zip.close reports error if nothing has been written to it
+		zip.close();
 	}
 
 	private void writeModuleSrcZipToOverallSrcZip(Zip zip, File srcZipFileForModule)
-- 
2.43.0

